[Bug] Flaky test: plugin/storage/badger/samplingstore

Sporadically fails, e.g. I was able to reproduce with this:
```
$ go test -count 10 ./plugin/storage/badger/samplingstore
```

```
--- FAIL: TestGetThroughput (0.08s)
    storage_test.go:58:
        	Error Trace:	/Users/ysh/dev/jaegertracing/jaeger/plugin/storage/badger/samplingstore/storage_test.go:58
        	            				/Users/ysh/dev/jaegertracing/jaeger/plugin/storage/badger/samplingstore/storage_test.go:142
        	            				/Users/ysh/dev/jaegertracing/jaeger/plugin/storage/badger/samplingstore/storage_test.go:47
        	Error:      	Not equal:
        	            	expected: []*model.Throughput{(*model.Throughput)(0x14000092270), (*model.Throughput)(0x140000922a0)}
        	            	actual  : []*model.Throughput(nil)

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,15 +1,2 @@
        	            	-([]*model.Throughput) (len=2) {
        	            	- (*model.Throughput)({
        	            	-  Service: (string) (len=6) "my-svc",
        	            	-  Operation: (string) (len=2) "op",
        	            	-  Count: (int64) 0,
        	            	-  Probabilities: (map[string]struct {}) <nil>
        	            	- }),
        	            	- (*model.Throughput)({
        	            	-  Service: (string) (len=7) "our-svc",
        	            	-  Operation: (string) (len=3) "op2",
        	            	-  Count: (int64) 0,
        	            	-  Probabilities: (map[string]struct {}) <nil>
        	            	- })
        	            	-}
        	            	+([]*model.Throughput) <nil>

        	Test:       	TestGetThroughput
```

--- a/plugin/storage/badger/samplingstore/storage.go
+++ b/plugin/storage/badger/samplingstore/storage.go
@@ -18,6 +18,7 @@ import (
 	"bytes"
 	"encoding/binary"
 	"encoding/json"
+	"fmt"
 	"time"
 
 	"github.com/dgraph-io/badger/v3"
@@ -95,7 +96,7 @@ func (s *SamplingStore) GetThroughput(start, end time.Time) ([]*model.Throughput
 			if err != nil {
 				return err
 			}
-			if t.Before(end) || t.Equal(end) {
+			if t.After(start) && (t.Before(end) || t.Equal(end)) {
 				retSlice = append(retSlice, throughputs...)
 			}
 		}
@@ -251,6 +252,7 @@ func initalStartTime(timeBytes []byte) (time.Time, error) {
 	buf := bytes.NewReader(timeBytes)
 
 	if err := binary.Read(buf, binary.BigEndian, &usec); err != nil {
+		fmt.Println("error-------------------------------------")
 		return time.Time{}, err
 	}
 

--- a/plugin/storage/badger/samplingstore/storage.go
+++ b/plugin/storage/badger/samplingstore/storage.go
@@ -18,7 +18,6 @@ import (
 	"bytes"
 	"encoding/binary"
 	"encoding/json"
-	"fmt"
 	"time"
 
 	"github.com/dgraph-io/badger/v3"
@@ -252,7 +251,6 @@ func initalStartTime(timeBytes []byte) (time.Time, error) {
 	buf := bytes.NewReader(timeBytes)
 
 	if err := binary.Read(buf, binary.BigEndian, &usec); err != nil {
-		fmt.Println("error-------------------------------------")
 		return time.Time{}, err
 	}
 

--- a/plugin/storage/badger/samplingstore/storage.go
+++ b/plugin/storage/badger/samplingstore/storage.go
@@ -96,7 +96,7 @@ func (s *SamplingStore) GetThroughput(start, end time.Time) ([]*model.Throughput
 				return err
 			}
 
-			if (t.After(start) || t.Equal(start)) && (t.Before(end) || t.Equal(end)) {
+			if t.After(start) && (t.Before(end) || t.Equal(end)) {
 				retSlice = append(retSlice, throughputs...)
 			}
 		}

--- a/plugin/storage/badger/samplingstore/storage.go
+++ b/plugin/storage/badger/samplingstore/storage.go
@@ -95,8 +95,7 @@ func (s *SamplingStore) GetThroughput(start, end time.Time) ([]*model.Throughput
 			if err != nil {
 				return err
 			}
-
-			if t.After(start) && (t.Before(end) || t.Equal(end)) {
+			if t.Before(end) || t.Equal(end) {
 				retSlice = append(retSlice, throughputs...)
 			}
 		}

--- a/plugin/storage/badger/samplingstore/storage.go
+++ b/plugin/storage/badger/samplingstore/storage.go
@@ -96,7 +96,7 @@ func (s *SamplingStore) GetThroughput(start, end time.Time) ([]*model.Throughput
 				return err
 			}
 
-			if t.After(start) && (t.Before(end) || t.Equal(end)) {
+			if (t.After(start) || t.Equal(start)) && (t.Before(end) || t.Equal(end)) {
 				retSlice = append(retSlice, throughputs...)
 			}
 		}

--- a/plugin/storage/badger/samplingstore/storage.go
+++ b/plugin/storage/badger/samplingstore/storage.go
@@ -95,6 +95,7 @@ func (s *SamplingStore) GetThroughput(start, end time.Time) ([]*model.Throughput
 			if err != nil {
 				return err
 			}
+
 			if t.After(start) && (t.Before(end) || t.Equal(end)) {
 				retSlice = append(retSlice, throughputs...)
 			}

--- a/plugin/storage/badger/samplingstore/storage.go
+++ b/plugin/storage/badger/samplingstore/storage.go
@@ -91,7 +91,7 @@ func (s *SamplingStore) GetThroughput(start, end time.Time) ([]*model.Throughput
 			if err != nil {
 				return err
 			}
-			throughputs, err := decodeThroughtputValue(val)
+			throughputs, err := decodeThroughputValue(val)
 			if err != nil {
 				return err
 			}
@@ -225,7 +225,7 @@ func (s *SamplingStore) createThroughputKV(throughput []*model.Throughput, start
 	return key, bb, err
 }
 
-func decodeThroughtputValue(val []byte) ([]*model.Throughput, error) {
+func decodeThroughputValue(val []byte) ([]*model.Throughput, error) {
 	var throughput []*model.Throughput
 
 	err := json.Unmarshal(val, &throughput)

--- a/plugin/storage/badger/samplingstore/storage_test.go
+++ b/plugin/storage/badger/samplingstore/storage_test.go
@@ -45,11 +45,13 @@ func TestInsertThroughput(t *testing.T) {
 
 func TestGetThroughput(t *testing.T) {
 	runWithBadger(t, func(t *testing.T, store *SamplingStore) {
-		start := time.Now().Truncate(time.Millisecond)
+		start := time.Now()
 		expected := []*samplemodel.Throughput{
 			{Service: "my-svc", Operation: "op"},
 			{Service: "our-svc", Operation: "op2"},
 		}
+
+		assert.Eventually(t, func() bool { return true }, time.Second, time.Microsecond*10)
 		err := store.InsertThroughput(expected)
 		require.NoError(t, err)
 

--- a/plugin/storage/badger/samplingstore/storage_test.go
+++ b/plugin/storage/badger/samplingstore/storage_test.go
@@ -53,7 +53,6 @@ func TestGetThroughput(t *testing.T) {
 		err := store.InsertThroughput(expected)
 		require.NoError(t, err)
 
-		time.Sleep(10 * time.Millisecond)
 		actual, err := store.GetThroughput(start, start.Add(time.Second*time.Duration(10)))
 		require.NoError(t, err)
 		assert.Equal(t, expected, actual)
@@ -121,7 +120,7 @@ func TestDecodeThroughtputValue(t *testing.T) {
 
 	marshalBytes, err := json.Marshal(expected)
 	require.NoError(t, err)
-	acrual, err := decodeThroughtputValue(marshalBytes)
+	acrual, err := decodeThroughputValue(marshalBytes)
 	require.NoError(t, err)
 	assert.Equal(t, expected, acrual)
 }

--- a/plugin/storage/badger/samplingstore/storage_test.go
+++ b/plugin/storage/badger/samplingstore/storage_test.go
@@ -45,7 +45,7 @@ func TestInsertThroughput(t *testing.T) {
 
 func TestGetThroughput(t *testing.T) {
 	runWithBadger(t, func(t *testing.T, store *SamplingStore) {
-		start := time.Now().Truncate(time.Microsecond)
+		start := time.Now()
 		expected := []*samplemodel.Throughput{
 			{Service: "my-svc", Operation: "op"},
 			{Service: "our-svc", Operation: "op2"},
@@ -54,7 +54,7 @@ func TestGetThroughput(t *testing.T) {
 		err := store.InsertThroughput(expected)
 		require.NoError(t, err)
 
-		actual, err := store.GetThroughput(start, start.Add(time.Second*time.Duration(10)))
+		actual, err := store.GetThroughput(start.Add(-time.Millisecond), start.Add(time.Second*time.Duration(10)))
 		require.NoError(t, err)
 		assert.Equal(t, expected, actual)
 	})

--- a/plugin/storage/badger/samplingstore/storage_test.go
+++ b/plugin/storage/badger/samplingstore/storage_test.go
@@ -50,7 +50,6 @@ func TestGetThroughput(t *testing.T) {
 			{Service: "my-svc", Operation: "op"},
 			{Service: "our-svc", Operation: "op2"},
 		}
-
 		err := store.InsertThroughput(expected)
 		require.NoError(t, err)
 

--- a/plugin/storage/badger/samplingstore/storage_test.go
+++ b/plugin/storage/badger/samplingstore/storage_test.go
@@ -45,7 +45,7 @@ func TestInsertThroughput(t *testing.T) {
 
 func TestGetThroughput(t *testing.T) {
 	runWithBadger(t, func(t *testing.T, store *SamplingStore) {
-		start := time.Now()
+		start := time.Now().Truncate(time.Millisecond)
 		expected := []*samplemodel.Throughput{
 			{Service: "my-svc", Operation: "op"},
 			{Service: "our-svc", Operation: "op2"},
@@ -53,6 +53,7 @@ func TestGetThroughput(t *testing.T) {
 		err := store.InsertThroughput(expected)
 		require.NoError(t, err)
 
+		time.Sleep(10 * time.Millisecond)
 		actual, err := store.GetThroughput(start, start.Add(time.Second*time.Duration(10)))
 		require.NoError(t, err)
 		assert.Equal(t, expected, actual)

--- a/plugin/storage/badger/samplingstore/storage_test.go
+++ b/plugin/storage/badger/samplingstore/storage_test.go
@@ -45,13 +45,12 @@ func TestInsertThroughput(t *testing.T) {
 
 func TestGetThroughput(t *testing.T) {
 	runWithBadger(t, func(t *testing.T, store *SamplingStore) {
-		start := time.Now()
+		start := time.Now().Truncate(time.Microsecond)
 		expected := []*samplemodel.Throughput{
 			{Service: "my-svc", Operation: "op"},
 			{Service: "our-svc", Operation: "op2"},
 		}
 
-		assert.Eventually(t, func() bool { return true }, time.Second, time.Microsecond*10)
 		err := store.InsertThroughput(expected)
 		require.NoError(t, err)
 

